apps/developer-portal/src/components/roles/create-role-wizard/role-basics.tsx

```diff
@@ -100,14 +110,43 @@ export const RoleBasics: FunctionComponent<RoleBasicProps> = (props: RoleBasicPr
                 "urn:ietf:params:scim:api:messages:2.0:SearchRequest"
             ],
             startIndex: 1
-        }
+        };
 
         searchRoleList(searchData)
             .then((response) => {
                 setIsValidRoleName(response?.data?.totalResults === 0);
             });
     };
 
+    /**
+     * The following function change of the user stores.
+     *
+     * @param values
+     */
+    const handleDomainChange = (values: Map<string, FormValue>) => {
+        const domain: string = values.get("domain").toString();
+        setUserStore(domain);
+    };
+
+    /**
+     * The following function validates role name against the user store regEx.
+     *
+     * @param roleName - User input role name
+     */
+    async function validateRoleNamePattern(roleName: string) {
```

Comment:
Shall we use an es6 async function here? Check other places as well.

```suggestion
    const validateRoleNamePattern = async (roleName: string) => {
```

Reply:
Resolved with 153fa58